### ISSUE TITLE
Make esgf search more modular

### ISSUE THREAD

**Issue by ScottWales** [coecms/clef#38]
ESGF search is a bit of a mess, make it more simple to use so it can be matched against the MAS database as we like

**Comment by paolap** [comment]
I started doing an alternative local_query to give something to work with with the "pipeline".
I've pushed this to the xrelease branch which is a copy of the current master plus these new functions.
I'm quite sure there's a more efficient way to do this 

I'm doing this in two steps:
out=session.query(c5_metadata_dataset_link],C5Dataset).join(C5Dataset).filter_by(**kwargs) out1=out.join(Path).join(ExtendedMetadata).filter(ExtendedMetadata.variable == var)
This is the CMIP5 example, for CMIP6 I do the same but I don't need to filter for the variable.

I'm not even sure the first step is necessary but that's the only way I tried and would return all the DRS bits: ensemble, experiment, frequency etc. and the path
After this I call a function that return a list of dictionaries for every file, then I have a different function that return a list of dictionaries but this time with a dicitonary for each simualtion a 'filenames' key listing all the files and a 'periods' key listing all the ranges.
I'm working on changing the periods to two fields from_date to_date.

Then I want to add some checks on the arguments a user can pass so we can warn a user when it pass keys/values which don't exist.
